package/kedro_viz/api/graphql.py

```diff
@@ -257,6 +263,88 @@ def run_tracking_data(
 schema = strawberry.Schema(query=Query, subscription=Subscription)
 
 
+@strawberry.input
+class RunInput:
+    """Run input to update bookmark, title and notes"""
+
+    bookmark: Optional[bool] = None
+    title: Optional[str] = None
+    notes: Optional[str] = None
+
+
+@strawberry.type
+class UpdateRunDetailsSuccess:
+    """Response type for sucessful update of runs"""
+
+    run_details: JSONObject
+
+
+@strawberry.type
+class BadInputType:
+    """Response type for failed update of runs"""
+
+    run_id: ID
+    error_message: str
+
+
+Response = strawberry.union(
+    "UpdateRunDetailsResponse", (UpdateRunDetailsSuccess, BadInputType)
+)
+
+
+@strawberry.type
+class Mutation:
+    """Mutation to update run details with run inputs"""
+
+    @strawberry.mutation
+    def update_run_details(self, run_id: ID, run_input: RunInput) -> Response:
+        """Updates run details based on run inputs provided by user"""
+        session = data_access_manager.db_session
+        run_data = get_runs([run_id])
+        if not run_data:
+            return BadInputType(
+                run_id=run_id, error_message="Given run_id doesn't exist"
+            )
+        
+        run_input.title = run_data[0].title if not run_input.title else run_input.title
```

Comment:
Let's pull `run_data[0]` out into a variable. Better yet, you can do 

```
try:
    run_data = next(get_runs([run_id])
except StopIteration:
    return BadInputType(...)
```

Reply:
hi, i wasn't able to get this working. It was throwing me NoneType is not iterable error. I tried to fix it via stackoverflow but couldn't get it to work. Please feel free to take over and push the above.